instance/tests/models/test_openedx_instance.py

```diff
@@ -1602,3 +1603,47 @@ def test_clean_up_dns_records(self, mocks, *mock):
         # archive instance and verify no dns records
         instance.archive()
         self._verify_vm_dns_records('test.com', [])
+
+@ddt.ddt
+# prevent default rabbitmq server from being created
+@override_settings(DEFAULT_RABBITMQ_API_URL=None)
+@patch(
+    'instance.tests.models.factories.openedx_instance.OpenEdXInstance._write_metadata_to_consul',
+    return_value=(1, True)
+)
+class OpenEDXInstanceRabbitMQTestCase(TestCase):
+    """
+    Tests RabbitMQ related features in OpenEDXInstance
+    """
+
+    def setUp(self):
+        """
+        Remove all rabbitmq servers to ensure test isolation
+        """
+        RabbitMQServer.objects.filter(accepts_new_clients=True).delete()
+
+    def test_assign_rabbitmq_fails_when_no_available_servers(self, mock_consul):
+        """
+        Test that new OpenEDX instances only assigned RabbitMQ instances with accepts_new_clients=True.
+
+        We actually do this by verifying that assignment fails if all instances have accepts_new_clients=False
+        """
+        with self.assertRaises(RabbitMQServer.DoesNotExist):
+            instance = OpenEdXInstance.objects.create(sub_domain='sandbox.defaults')
+    
+    def test_assigns_available_server(self, mock_consul):
+        def new_rabbitmq(accepts_new_clients, suffix="default"):
+            return RabbitMQServer.objects.create(
+                accepts_new_clients=accepts_new_clients,
+                api_url=f"https://example-{suffix}.com/api",
+                admin_username="user",
+                admin_password="password",
+                name="name-{suffix}",
+                instance_host="example.com",
+            )
```

Comment:
Can't we turn this into a factory and put in `instance/tests/models/factories`?

Reply:
Fixed!